libs/langgraph/langgraph/utils/future.py

```diff
@@ -180,25 +186,32 @@ def _wrap_awaitable(awaitable: Awaitable[T]) -> Generator[None, None, T]:
 def run_coroutine_threadsafe(
     coro: Coroutine[None, None, T],
     loop: asyncio.AbstractEventLoop,
+    *,
+    lazy: bool,
     name: Optional[str] = None,
     context: Optional[contextvars.Context] = None,
 ) -> asyncio.Future[T]:
     """Submit a coroutine object to a given event loop.
 
     Return a asyncio.Future to access the result.
     """
-    future: asyncio.Future[T] = asyncio.Future(loop=loop)
 
-    def callback() -> None:
-        try:
-            chain_future(
-                _ensure_future(coro, loop=loop, name=name, context=context), future
-            )
-        except (SystemExit, KeyboardInterrupt):
-            raise
-        except BaseException as exc:
-            future.set_exception(exc)
-            raise
-
-    loop.call_soon_threadsafe(callback, context=context)
-    return future
+    if asyncio._get_running_loop() is loop:
```

Comment:
note this is function is documented "This is a low-level function intended to be used by event loops", so not private api